torch/csrc/jit/python/python_custom_class.cpp

```diff
@@ -14,15 +16,26 @@ py::object ScriptClass::__call__(py::args args, py::kwargs kwargs) {
   auto instance =
       Object(at::ivalue::Object::create(class_type_, /*numSlots=*/1));
   Function* init_fn = instance.type()->findMethod("__init__");
-  TORCH_CHECK(
-      init_fn,
-      fmt::format(
-          "Custom C++ class: '{}' does not have an '__init__' method bound. "
-          "Did you forget to add '.def(torch::init<...>)' to its registration?",
-          instance.type()->repr_str()));
+  auto input_args = std::move(args);
+  auto input_kwargs = std::move(kwargs);
+  if (!init_fn) {
+    if (auto methods = instance.type()->findOverloadedMethod("__init__")) {
+      auto resolved_init_method = match_overloaded_methods(
+          instance._ivalue(), "__init__", input_args, input_kwargs);
+      invokeScriptMethodFromPython(
+          resolved_init_method.value(), input_args, input_kwargs);
```

Comment:
I think you still need to `std::move` on `input_args` and `input_kwargs` for maximum performance even though you moved them once.

Same for the call on line 38.